Add EAR format for extracting artifacts

florentdupont opened the pull request:
Some format are accepted to be extracted but some other are not. I'd like to be able to extract `EAR` artifacts as well as zip, war or jar artifacts.

This pull request adds this functionality.

florentdupont committed:
add EAR format for extracting artifacts

KAllan357 commented:
Sorry, there was an outstanding pull-request that fixed up this section of the code. Could you merge this with that?

florentdupont commented:
Hi. Sorry I forgot to update from the master here.
I've updated my fork and I see that the contract has changed for WAR/JAR files in `extract_artifact!`.
I've added a comment on issue #142.
I used to use the `is_tarball` parameter to extract (or not) my artifacts, no matters if they are zip/war/jar/whatever-compressed-format : when `is_tarball` is false => artifact is copied, when `is_tarball` is true => artifact is uncompressed.
In my case, I'd like to include ear as a "zipped" file format that could be uncompressed in the extract_artifact! method (as well as jar and war) and assure it will be extracted if I set `is_tarball` to true.

florentdupont commented:
I'm removing this request since the modification made on #142 are incompatible with this PR.

We change the way we are deploying EAR/WAR/JAR internaly. It's better IMHO to let the application server extract the artifacts (I mean Java Archive such as EAR and WAR) on deployment. And I don't see the need to extract a JAR file neither.
I don't need this PR anymore.
